### PR TITLE
Address unused local bind.

### PR DESCRIPTION
**Pull request by jacereda**
This one probably requires closer inspection. I tried to leave debugging helpers commented, but maybe you prefer to leave them uncommented.

### CODE REVIEW

File: src/Emit.hs [hunk 1]
```diff
@@ -313,8 +312,6 @@ toC toCMode (Binder meta root) = emitterSrc (execState (visit startingIndent roo
                     -- This requires a bunch of extra machinery though, so this will do for now...
                     [var ++ periodOrArrow ++ "_tag == " ++ tagName caseTy (removeSuffix caseName)] ++
                       concat (zipWith (\c i -> tagCondition (var ++ periodOrArrow ++ "u." ++ removeSuffix caseName ++ ".member" ++ show i) "." (forceTy c) c) caseMatchers ([0..] :: [Int]))
-                      where notUnitX (XObj _ _ (Just UnitTy)) = False
```
Comment (scolsen) [inline src/Emit.hs:316]: Ah, this is possibly a bug--probably we want to filter Units from `caseMatchers` here. Mind adding a todo to double check this?

File: src/Eval.hs [hunk 2]
```diff
@@ -483,9 +483,7 @@ catcher ctx exception =
 
 specialCommandWith :: Context -> XObj -> SymPath -> [XObj] -> IO (Context, Either EvalError XObj)
 specialCommandWith ctx xobj path forms = do
-  let pathStrings = contextPath ctx
```
Comment (scolsen) [inline src/Eval.hs:486]: pathStrings use here is probably also a potential bug/cleanup--I think for almost all if not all of these lookups we should not be using `path` but rather:

`consPath (union (contextPath context) modulePrefixes) path`

where `modulePrefixes` are the list of prefixes in the given `SymPath` argument--this ensures we lookup the right things when we're evaluating something in some local context. I could be wrong about this, though @hellerve @eriksvedang

File: src/GenerateConstraints.hs [hunk 3]
```diff
@@ -89,8 +89,7 @@ genConstraints globalEnv root rootSig = fmap sort (gen root)
                                 trueType <- toEither (ty ifTrue) (ExpressionMissingType ifTrue)
                                 falseType <- toEither (ty ifFalse) (ExpressionMissingType ifFalse)
                                 let expected = XObj (Sym (SymPath [] "Condition in if value") Symbol) (info expr) (Just BoolTy)
-                                let lol = XObj (Sym (SymPath [] "lol") Symbol) (info expr) (Just BoolTy)
```
Comment (scolsen) [inline src/GenerateConstraints.hs:92]: heh, I always wondered about this :)

**Review by scolsen — COMMENTED** [review]
Generally looks good  to me!

Left a few comments regarding missing uses of bindings that I think are more likely bugs than not, but we don't need to address those here. 

Awesome cleanup!

File: src/GenerateConstraints.hs (hunk 3 above)
Comment (eriksvedang) [reply src/GenerateConstraints.hs:92]: 🤔

File: src/Emit.hs (hunk 1 above)
Comment (jacereda) [reply src/Emit.hs:316]: Done. Would it need `isUnit` or the one removed that just checks plain units?

File: src/GenerateConstraints.hs (hunk 3 above)
Comment (jacereda) [reply src/GenerateConstraints.hs:92]: Probably some test by @hellerve, introduced in 6e147598a.

File: src/Emit.hs (hunk 1 above)
Comment (scolsen) [reply src/Emit.hs:316]: iirc the list was a list of xobjs in this case, so `isUnit` will work but will need to be composed w/ `forceTy`, I think `filter (isUnit . forceTy)` should do it

File: src/Eval.hs (hunk 2 above)
Comment (eriksvedang) [reply src/Eval.hs:486]: Hmm I'm not sure without thinking more about it. Maybe create an issue for it? @scolsen

**Review by eriksvedang — APPROVED** [review]

**Pull request merged**